sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -742,12 +750,41 @@ object ScalaJSPluginInternal {
 
   // These settings will be filtered by the stage dummy tasks
   val scalaJSRunSettings = Seq(
+      scalaJSMainModuleInitializer := {
+        if (scalaJSUseMainModuleInitializer.value) {
+          mainClass.value.fold {
```

Comment:
No, I don't think so. It seems weird to me that one would want a different `mainClass` to compute `scalaJSMainModuleInitializer` versus the names displayed by `Running helloworld.Main` and other things like that.